### PR TITLE
UIF-286 limit rollover fscal years to future

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
## Purpose
* Limit selection of fiscal years during rollover to upcoming years;
* used react-query to load fiscal years;
https://issues.folio.org/browse/UIF-286

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by axelhunn — APPROVED** [review]

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            data: { fiscalYears: [{ id: 'fyId' }] },
+          };
+        },
+      });
+    });
+
+    const { result, waitFor } = renderHook(() => useRolloverFiscalYears('FY'), { wrapper });
+
+    await waitFor(() => {
+      return Boolean(result.current.fiscalYears);
```
Comment (aliaksei-chumakou) [inline src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:44]: @NikitaSedyx could you please look why it can't wait  this happening?

**Review by AndreiShumski2 — APPROVED** [review]

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const useRolloverFiscalYears = (series) => {
+  const ky = useOkapiKy();
+
+  const params = {
+    limit: LIMIT_MAX,
+    query: `series=="${series}" sortby periodStart`,
+  };
+  const { data: { fiscalYears } = {}, isLoading } = useQuery(
+    ['finance', 'fiscal-years', series],
+    () => ky(`${FISCAL_YEARS_API}?${queryString.stringify(params)}`).json(),
```
Comment (NikitaSedyx) [inline src/Ledger/RolloverLedger/useRolloverFiscalYears.js:18]: mb `{ searchParams: params }`?

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js (hunk 1 above)
Comment (NikitaSedyx) [reply src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:44]: should be ok, try to print `result.error` before await, mb there is an error in hook

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const { result, waitFor } = renderHook(() => useRolloverFiscalYears(), { wrapper });
+
+    await waitFor(() => {
+      return !result.current.isLoading;
+    });
+
+    expect(result.current.fiscalYears).toBeUndefined();
+  });
+
+  xit('should return response array', async () => {
```
Comment (AndreiShumski2) [inline src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:30]: ^^ `xit`

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js (hunk 1 above)
Comment (aliaksei-chumakou) [reply src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:44]: no luck, 
```
    console.log
      result.error undefined
```

Comment (aliaksei-chumakou) [reply src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:42]: I give up on this, it works in app, but not in tests. maybe something messed up with mocking or rendering, but result doesn't change.

Comment (NikitaSedyx) [reply src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:42]: I can checkout and try later

**Review by NikitaSedyx — APPROVED** [review]

File: src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js (hunk 1 above)
Comment (aliaksei-chumakou) [reply src/Ledger/RolloverLedger/useRolloverFiscalYears.test.js:42]: thanks a lot!

**Pull request merged**